modules/core/shared/src/test/scala/org/legogroup/woof/EnclosingClassSuite.scala

```diff
@@ -0,0 +1,51 @@
+package org.legogroup.woof
+
+import cats.Applicative
+import cats.effect.{Clock, IO}
+import cats.instances.string
+import cats.kernel.Monoid
+import cats.syntax.all.*
+import munit.{CatsEffectSuite, FunSuite}
+import org.legogroup.woof.Filter.given_Monoid_Filter
+import org.legogroup.woof.Logger.*
+
+import java.time.format.DateTimeFormatter
+import java.time.{Instant, ZoneId}
+import scala.concurrent.duration.*
+class EnclosingClassSuite extends CatsEffectSuite:
+
+  class ThisClassNameIsReallyLongAndWillTriggerParentPackageAbbreviation:
+
+    def testProgram(using Logger[IO]): IO[Unit] =
+      for
+        _ <- Logger[IO].info("Info message")
+        _ <- Logger[IO].debug("Debug message")
+        _ <- Logger[IO].warn("Warning message")
+        _ <- Logger[IO].error("Error message")
+      yield ()
+  end ThisClassNameIsReallyLongAndWillTriggerParentPackageAbbreviation
+
+  given Filter = Filter.everything
+  given Printer = NoColorPrinter(testFormatTime)
+
+  test("class names abbreviated when too long") {
+    val expected = "o.l.w.E.ThisClassNameIsReallyLongAndWillTriggerParentPackageAbbreviation"
+    val testClass = new ThisClassNameIsReallyLongAndWillTriggerParentPackageAbbreviation
+    for
+      stringWriter     <- newStringWriter
+      given Logger[IO] <- DefaultLogger.makeIo(stringWriter)(using leetClock)
+      _                <- testClass.testProgram
+      str              <- stringWriter.get
+    yield assert(str.contains(expected))
+  }
+
+  test("terminates when package depth is really deep") {
+    assertEquals(EnclosingClass("never.gonna.give.you.up", 1).printableName, "n.g.g.y.up")
+  }
+
+  test("does not abbreviate package if too short") {
+    assertEquals(EnclosingClass("Too.$hort", 10).printableName, "Too.$hort")
+  }
+
```

Comment:
```suggestion
  test("partially chomped") {
    assertEquals(EnclosingClass("never.gonna.give.you.up", 10).printableName, "n.g.give.you.up")
  }
```

Reply:
I am thinking that the `printableName` should be reduced to as close to 10 characters as possible, without touching the final package name.  This seems a little surprising to a caller, and is a result of my observation [here](https://github.com/LEGO/woof/pull/27#discussion_r778148445) where we might want to use a delimiter before we compare the length while unfolding.